src/scheduler/annualReviewStartEmailScheduler.ts

```diff
@@ -0,0 +1,211 @@
+  if (findListItemsResult.error) {
+    logger.error(`Unable to send emails.  Unable to retrieve List Items for Lists ${listIds}`);
+
+  } else if (findListItemsResult.result) {
+    const listItemsForAllLists = findListItemsResult.result;
+    for (const list of lists) {
+      // get list items eligible for annual review
+      const listItemsEligibleForAnnualReview = listItemsForAllLists.filter(listItem => listItem.listId === list.id);
+      logger.info(`checking list ${list.id} ${context.daysBeforeAnnualReviewStart} ${context.datePart}
+      before annual review date ${list.nextAnnualReviewStartDate} with ${listItemsEligibleForAnnualReview.length}
+      listItems [${listItemsEligibleForAnnualReview.map(listItem => listItem.id)}]
+      and publishers [${list.jsonData.publishers}]`);
+
+      if (listItemsForAllLists.length > 0) {
+        await emailPosts(listItemsEligibleForAnnualReview, list, context);
+
+        if (context.daysBeforeAnnualReviewStart === 0) {
+          // update list items and email providers to confirm annual review start
+          const updatedListItems: ListItem[] = await updateAnnualReview(
+            listItemsEligibleForAnnualReview,
+            Status.CHECK_ANNUAL_REVIEW,
+            ListItemEvent.ANNUAL_REVIEW_STARTED,
+            "startAnnualReview",
+            AuditEvent.ANNUAL_REVIEW);
+          const listItemsNotUpdated = listItemsEligibleForAnnualReview.filter(listItem => {
+            return updatedListItems.map(updatedListItem => updatedListItem.id).includes(listItem.id);
+          })
+          if (listItemsNotUpdated) {
+            logger.info(`List items ${listItemsNotUpdated.map(listItem => listItem.id)} could not be updated`);
+          }
+
+          await emailProviders(updatedListItems, list, context);
+        }
+      }
+    }
+  }
+}
+
+interface AnnualReviewStartDateContext {
+  annualReviewStartDate: Date;
+  daysBeforeAnnualReviewStart: number;
+  datePart: string;
+}
+
+function getAnnualReviewStartDateContexts(todayDateString: string): AnnualReviewStartDateContext[] {
+  const today = getTodayDate(todayDateString);
+
+  const annualReviewStartDateContexts: AnnualReviewStartDateContext[] = [
+    {
+      annualReviewStartDate: addDays(today, ONE_MONTH_AWAY),
+      daysBeforeAnnualReviewStart: ONE_MONTH_AWAY,
+      datePart: "day",
+    },
+    {
+      annualReviewStartDate: addDays(today, ONE_WEEK_AWAY),
+      daysBeforeAnnualReviewStart: ONE_WEEK_AWAY,
+      datePart: "day",
+    },
+    {
+      annualReviewStartDate: addDays(today, ONE_DAY_AWAY),
+      daysBeforeAnnualReviewStart: ONE_DAY_AWAY,
+      datePart: "day",
+    },
+    {
+      annualReviewStartDate: today,
+      daysBeforeAnnualReviewStart: 0,
+      datePart: "day",
+    },
+  ];
+  return annualReviewStartDateContexts;
+}
+
+export async function sendAnnualReviewStartEmails(todayDateString: string): Promise<void> {
```

Comment:
This is interesting, so is the `sendAnnualReviewStartEmails` function used outside of this file by an external process? I was trying to find out why the `export` keyword was here and I couldn't see it in the project so I'm assuming some cool infrastructure magic is happening with this function?